t/api_calls.t

```diff
@@ -259,6 +258,100 @@ describe "A duo api client" => sub {
             is($req2->method, 'GET');
         };
     };
+
+    describe "make_request method" => sub {
+        my @calculate_backoff_calls;
+        my @sleep_calls = [];
+        my @response_codes;
+        my $req = mock();
+        $req->stubs(
+            as_string => "I am a request",
+        );
+
+        before each => sub {
+            @calculate_backoff_calls = ();
+            @sleep_calls = ();
+            @response_codes = ();
+
+            $mock_response->stubs(
+                code => sub {
+                    return shift(@response_codes) || 200;
+                },
+            );
+
+            $sut->stubs(
+                calculate_backoff => sub {
+                    push @calculate_backoff_calls, $_[1];
+                    return $_[1];
+                },
+            );
+
+            Time::HiRes->stubs(
+                sleep => sub {
+                    push @sleep_calls, $_[0];
+                },
+            );
+        };
+
+        it "makes a single call when 200 response" => sub {
+            @response_codes = (200);
+            my $resp = $sut->make_request($req);
+
+            is($resp, $mock_response);
+            is(@captured_requests, 1);
+            is($last_captured_req, $req);
+            is(@calculate_backoff_calls, 0);
+            is(@sleep_calls, 0);
+        };
+
+        it "retries after being rate limited" => sub {
+            @response_codes = (429, 200);
+            my $resp = $sut->make_request($req);
+
+            is($resp, $mock_response);
+            cmp_deeply(\@captured_requests, [$req, $req]);
+
+            # Calculate backoff was mocked to return whatever retry attempt this was
+            cmp_deeply(\@calculate_backoff_calls, [0]);
+            cmp_deeply(\@sleep_calls, [0]);
+        };
+
+        it "will only make a max of 7 requests before stopping retries" => sub {
+            @response_codes = (429, 429, 429, 429, 429, 429, 429, 429, 429,
```

Comment:
`@response_codes = (429) x 15` will be a bit more readable here

Reply:
👍 
I actually probably should make this x6 or x7.  There is no specific reason for 15, I just wanted to make sure there was more than enough to handle every request.